src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -3868,6 +3868,13 @@ void AuraEffect::HandleModCastingSpeed(AuraApplication const* aurApp, uint8 mode
 
     Unit* target = aurApp->GetTarget();
 
+    // Do not apply such auras in normal way
+    if (GetAmount() >= 1000)
+    {
+        target->SetInstantCast(apply);
```

Comment:
It's not possible in 3.3.5 (very few of such auras). If theoretically this could happen, we would need to loop speed_not_stack auras to see if there is other such aura and prevent unsetting